github/table_github_team.go

```diff
@@ -0,0 +1,177 @@
+	return &plugin.Table{
+		Name:        "github_team",
+		Description: "GitHub Teams in a given organization. GitHub Teams are groups of organization members that reflect your company or group's structure with cascading access permissions and mentions.",
+		List: &plugin.ListConfig{
+			KeyColumns:        plugin.SingleColumn("organization"),
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			Hydrate:           tableGitHubTeamList,
+		},
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"organization", "slug"}),
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			Hydrate:           tableGitHubTeamGet,
+		},
+		Columns: gitHubOrgTeamColumns(),
+	}
+}
+
+//// LIST FUNCTION
+
+func tableGitHubTeamList(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	client := connect(ctx, d)
+
+	opt := &github.ListOptions{PerPage: 100}
+
+	quals := d.KeyColumnQuals
+	org := quals["organization"].GetStringValue()
+
+	limit := d.QueryContext.Limit
+	if limit != nil {
+		if *limit < int64(opt.PerPage) {
+			opt.PerPage = int(*limit)
+		}
+	}
+
+	type ListPageResponse struct {
+		teams []*github.Team
+		resp  *github.Response
+	}
+
+	listPage := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+		teams, resp, err := client.Teams.ListTeams(ctx, org, opt)
+		return ListPageResponse{
+			teams: teams,
+			resp:  resp,
+		}, err
+	}
+
+	for {
+		listPageResponse, err := plugin.RetryHydrate(ctx, d, h, listPage, &plugin.RetryConfig{ShouldRetryError: shouldRetryError})
+
+		if err != nil {
+			return nil, err
+		}
+
+		listResponse := listPageResponse.(ListPageResponse)
+		teams := listResponse.teams
+		resp := listResponse.resp
+
+		for _, i := range teams {
+			if i != nil {
+				d.StreamListItem(ctx, i)
+			}
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+
+		opt.Page = resp.NextPage
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func tableGitHubTeamGet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var org, slug string
+	if h.Item != nil {
+		team := h.Item.(*github.Team)
+		slug = *team.Slug
+
+		// Present for the authenticated user repos.
+		if team.Organization != nil {
+			org = *team.Organization.Login
+		} else { // Not present listing org repos.
+			org = d.KeyColumnQuals["organization"].GetStringValue()
+		}
+	} else {
+		org = d.KeyColumnQuals["organization"].GetStringValue()
+		slug = d.KeyColumnQuals["slug"].GetStringValue()
+	}
+
+	client := connect(ctx, d)
+
+	type GetResponse struct {
+		team *github.Team
+		resp *github.Response
+	}
+
+	getDetails := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+		detail, resp, err := client.Teams.GetTeamBySlug(ctx, org, slug)
```

Comment:
Here and in the other tables I use the APIs based on `slug` as getting info by `id` is listed as `legacy` by the [API docs](https://docs.github.com/en/rest/teams/teams#get-a-team-legacy).

Reply:
@japborst you are absolutely correct to use the APIs based on `slug`, however since the majority of our queries and API calls are based on `id` (including the example queries), we might consider moving it as a part of breaking change.

Reply:
@misraved perhaps we should then migrate all of them 🤔 How do you suggest moving forward?

Reply:
@japborst The migration definitely makes sense since we should always look to update to the latest APIs.

Few points to be addressed:
- How many tables need to be updated to use the latest API format?
- We need to update the documents of the corresponding tables to include the latest changes.
-  The tables should still support querying by `id` and `slug` (tables should continue to require `organization` and at least one of `id` or `slug`, if both are passed in, prioritize `slug`). Updating the `github_my_team` table to use only `slug` seems like a hurried change since we would like to continue the support for `id` before we can phase it out completely.

Would be great to add those changes to this PR.

Reply:
There are definitely more cases from the API docs, but looking at the code `*ById(...)` actually doesn't give us other results, so this is the only case 🎉 

I've pushed a commit that handles both cases and tested this works.

Supporting both is a little ugly, as all fields become optional to support both cases. Have a look at the get key columns. That also means less intuitive error messages.

Releasing without backwards compatibility would be preferred 😅

Reply:
After discussing with @misraved and reviewing the code changes, I think it's probably ok to not make these changes backward compatible and to just support getting a team in either table by organization + slug. As long as we're not removing or changing the data for any columns, the changes originally proposed are ok.

Reply:
Clear 👍  The exposed data indeed doesn't change. The team `id` and `organization_id` are still there, for people relying on it.

I've reverted my last commit